docs/Features/XMLReferencesFeatures.md

```diff
@@ -0,0 +1,122 @@
+
+```json
+  "xml.references": [
+    {
+      "pattern": "**/*.xml",
+      "expressions": [
+        {
+          "from": "xref/@linkend",
+          "to": "@id"
+        }
+      ]
+    }
+  ]
+```
+
+After saving this settings, you will enjoy with `completion, definition, highlighting` with `xref` docbook:
+
+![XML References](../images/Features/XMLReferences.gif)
+
+The `xml.references` settings is an array of reference which is declared with:
+
+ * `pattern`:  matches the files that reference declared with `expressions` applies to. See [glob syntax](https://docs.oracle.com/javase/tutorial/essential/io/fileOps.html#glob) for more information about the pattern syntax. 
+ * `expressions`: array of reference expression:
+    * `prefix (optional)`: the prefix to use (ex : '#')  for from.
+    * `from`: the from reference DOM node declared with XPath.
+    * `to`: the to reference DOM node declared with XPath.
+ 
+### prefix
+
+Given this [TEI](https://tei-c.org/) XML file sample:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-model href="http://www.tei-c.org/release/xml/tei/custom/schema/relaxng/tei_lite.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
+<TEI xmlns="http://www.tei-c.org/ns/1.0">
+  <teiHeader>  
+    <fileDesc>
+      <titleStmt>
+        <title>Title</title>
+      </titleStmt>
+      <publicationStmt>
+        <p>Publication information</p>  
+      </publicationStmt>
+      <sourceDesc>
+        <p>Information about the source</p>
+      </sourceDesc>
+    </fileDesc>
+  </teiHeader>
+  <text>
+    <body xml:id="body-id">
+      <p xml:id="p-id" >Some text here.</p>
+      <anchor corresp="#body-id"></anchor>
+    </body>
+  </text>
+</TEI>
+```
+
+In this sample, `corresp` attribute in `<anchor corresp="#body-id"></anchor>` references the `body-id` (without `#`) declared in `<body xml:id="body-id">`. It means that the `corresp` attribute value (with `#`) reference `@xml:id` attribute (without `#`). To support that, you can configure settings by using `prefix`:
+
+```json
+"xml.references": [
+    {
+      "pattern": "**/*.xml",
+      "expressions": [
+        {
+          "prefix": "#",
+          "from": "@resp",
+          "to": "persName/@xml:id"
+        },
+        {
+          "prefix": "#",
+          "from": "@corresp",
+          "to": "@xml:id"
+        }
+      ]
+    }
+  ]
+```
+
+
+### Limitation
+
+XML references have some limitation:
+
+ * `references works only for a given XML file`: it is not possible to reference some DOM nodes coming from another XML files. However if the file uses include (like xi:include) the reference should work for only the master file which include another file, but it will not work for the included file.
```

Comment:
... the reference will only work in the file which has the include statement, and not in the file being included.

Reply:
fixed